cacheops/simple.py

```diff
@@ -46,8 +48,20 @@ def wrapper(*args, **kwargs):
                 cache_key = 'c:' + _get_key(func, args, kwargs, extra)
                 try:
                     result = self.get(cache_key)
+
                 except CacheMiss:
-                    result = func(*args, **kwargs)
+                    if USE_LOCK:
+                        # This will block in all calls but the
+                        # first to avoid dog-pile effect.
+                        with redis_lock.Lock(self.conn, cache_key):
```

Comment:
`getattr(self, 'conn', None)` will work, but it's a dirty hack. Just use `redis_client` directly for locks here for now, I'll think how this could be done better some time in the future.